src/main/java/com/minecolonies/coremod/client/gui/WindowResourceList.java

```diff
@@ -77,6 +124,43 @@ public WindowResourceList(final int colonyId, final BlockPos buildingPos)
         this.builder = null;
     }
 
+    /**
+     * Checks the resources in the warehouse to check for any resources required by the builder,
+     * only does anything when the warehouse position is provided.
+     */
+    private void pullResourcesFromWarehouse()
+    {
+        if (warehousePos == null)
+        {
+            return;
+        }
+
+        Map<Item, BuildingBuilderResource> resourceMap = resources.stream().collect(Collectors.toMap(ItemStorage::getItem, v -> v));
+
+        warehouseSnapshot = new HashMap<>();
```

Comment:
can't we always just initiate this with an empty map?

Reply:
In the constructor it attempts to load the snapshot if any NBT data is present, else it's an empty hashmap, however this resets the snapshot in order to clear out a potential existing snapshot to then proceed to populate it with new data.

Reply:
ah kk